libvast/src/system/configuration.cpp

```diff
@@ -48,6 +48,17 @@ template <concrete_type T>
 struct has_extension_type
   : std::is_base_of<arrow::ExtensionType, typename T::arrow_type> {};
 
+/// Translates an environment variable to a config key.
+/// A '.' in the YAML config maps to '_' in the env key. A literal '_' in a
+/// key requires double-escaping, i.e., "__".
```

Comment:
Can we re-use the mechanism from Dynaconf here, which I think is more widely used?

- `__` translates to a nesting layer: `.`
- `_` translates to a dash: `-`

Note that this does not apply to the application name prefix:

- `VAST_ENDPOINT` translates to `vast.endpoint`
- `VAST_IMPORT__BATCH_TIMEOUT` translates to `vast.import.batch-timeout`

Reply:
Agreed, this is the way the user would expect this to work.